chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,248 @@
+import cupy as cp
+import numpy as np
+
+import chainer.functions as F
+
+
+class DiscriminativeMarginBasedClusteringLoss:
+    """ Discriminative margin based clustering loss function
+
+    This is the implementation of following paper: 
+    https://arxiv.org/pdf/1802.05591.pdf
+
+    In segmentation, one of the biggest problem is having noise at the output 
+    of trained network.
+    For cross-entropy based approaches, if the pixel value is wrong,
+    the loss value will be same independent from wrong pixel's location.
+    However, for segmentation, even though network gives wrong pixel output, 
+    it is desirable to have it
+    as close as possible to the original position.
+    By applying discriminative loss function, groups of segmentation instances 
+    can be moved together.
+
+    This loss function calculates three different parameters:
```

Comment:
We follow [Google style guide](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html#google-vs-numpy) in writing documents by using Napoleon, which is an extension of Sphinx that allows us to write documents following the guideline. So, could you write in that way?